[Website] : Improve Lighthouse Score

Akshun-01 opened the pull request:
**Description**
Change video format from .gif to .webp
This PR fixes #973 

**Notes for Reviewers**
I would delete the original .gif files during final push, for now I have left them as it is!

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

abhijeetgauravm commented:
Hey @Akshun-01  Let's discuss this on website call today at 6:30 PM IST / 7 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima).

Yashsharma1911 commented:
@Akshun-01 Thankyou for raising a draft PR, can you share the progress here?

Akshun-01 commented:
## 2nd Update : 
### Changed the images (only among the ones pointed by analysis) from ```.png to .webp``` format

### Before
![image](https://github.com/meshery/meshery.io/assets/93793691/3983995d-d3dc-45d6-8701-ae53f8f6285e)

### After
![image](https://github.com/meshery/meshery.io/assets/93793691/d1de39c2-47cb-4be9-a623-0a940ff2e736)

Akshun-01 commented:
Hey @Yashsharma1911, I was confused about this one.
Do I need to change the size(dimensions) of the original image?
Won't it be troublesome, could you please explain this point!

![image](https://github.com/meshery/meshery.io/assets/93793691/1b843abc-5751-4bb7-b364-56941a25d412)

Yashsharma1911 commented:
There can be a couple of solutions you might like to read this blog https://developer.chrome.com/docs/lighthouse/performance/uses-responsive-images, I guess it's about serving image that is appropriate to screen it can differ based on resolution

I'll recommend to not hander the original image rather different size version of original images and you can use `srcset` attribute to pass multiple images for resolution. Lemme know your findings

Yashsharma1911 commented:
> ## 2nd Update :
> ### Changed the images (only among the ones pointed by analysis) from `.png to .webp` format
> ### Before
> ![image](https://private-user-images.githubusercontent.com/93793691/292870728-3983995d-d3dc-45d6-8701-ae53f8f6285e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NjI1MTgsIm5iZiI6MTcwMzY2MjIxOCwicGF0aCI6Ii85Mzc5MzY5MS8yOTI4NzA3MjgtMzk4Mzk5NWQtZDNkYy00NWQ2LTg3MDEtYWU1M2Y4ZjYyODVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA3MzAxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE2OTdmNzEwMWU4Mzk3ZThkNzhhZGIxNGFjZDE1Y2Q0ZDQ4YWIxZWNjYTE2ZWVhODQ5YjQ5NzI3MWNiYjhjM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f_OYRzY23nXdoLzePGTuIpfrga3ErSdcd8QBXImdEn0)
> 
> ### After
> ![image](https://private-user-images.githubusercontent.com/93793691/292870750-d1de39c2-47cb-4be9-a623-0a940ff2e736.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NjI1MTgsIm5iZiI6MTcwMzY2MjIxOCwicGF0aCI6Ii85Mzc5MzY5MS8yOTI4NzA3NTAtZDFkZTM5YzItNDdjYi00YmU5LWE2MjMtMGE5NDBmZjJlNzM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA3MzAxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZWQ2NjRhYTBlYjZiOTU1NDRmN2IzYTAzYWU4MjNjZTY5ZjE3ZjIyMzhlN2U0NGM0MzhhOWY4OTZkYTk4MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h3W9CnSoqxWlWyfrwnTWCQo9gBPHeFw2gK5fNgY8e9w)

This is Good 👏

Akshun-01 commented:
Helo!
## [UPDATE] : Performance is at 72 (slight decrease idky)

### - Changed some other heavy images from png to webp 
### - Resized the main meshmap gif and added responsive alternative for it

Now, these are the main problem points being highlighted: 
![image](https://github.com/meshery/meshery.io/assets/93793691/73cb1ad9-7421-422c-aa74-a63ff77a0f1b)

So, I think next we can focus on 
- text compression
- prioritising what to ship first (will need to look into it in detail)

Akshun-01 commented:
@Yashsharma1911, I don't think we can improve it much now, as it is already touching 70 now. It would need lot of optimizations to reach to 80. So, how should we proceed with the issue?

Yashsharma1911 commented:
> @Yashsharma1911, I don't think we can improve it much now, as it is already touching 70 now. It would need lot of optimizations to reach to 80. So, how should we proceed with the issue?

Thankyou @Akshun-01 it would be nice if you can present this in tomorrows website meeting

Yashsharma1911 commented:
@Akshun-01 any update here?

Raj100679 commented:
@Yashsharma1911 I don't have much experience at reviewing , but I will need to check what the issue is, then I will review it at night, Thanks

Akshun-01 commented:
Hey @Yashsharma1911 , I tried some other suggestions given by the report. But now rather than increasing it is rather going the opposite way.
Also the average performance was way better after the second commit ( ie after changing only the formats) 
After I changed the image sizes and added multiple images for `responsive image feedback` it is giving inconsistent reading everytime I check. (ranging from 50 - 75 ).
So, I doubt making that change was a wise choice. It is my first time making these optimizations, so I also don't know if we should consider all feedback or do the relevant ones only. 
Could you please check and see how should I proceed with this now. 
Thanks!^^

@Raj100679  would appreciate your feedback and insight on this!

Raj100679 commented:
@Akshun-01 Sorry I am stuck with some other work, @Yashsharma1911 Can you check it

leecalcote commented:
Thanks for working through this, @Akshun-01.

sudhanshutech commented:
@Akshun-01 still working on this?

sudhanshutech commented:
> @Akshun-01 still working on this?

Got any solution?

sudhanshutech commented:
following up again, @Akshun-01 are you still working on this?

Akshun-01 commented:
Hi @sudhanshutech, sorry for the late reply. 

Actually, I was waiting for someone to review the already-made changes. 
Also as I mentioned earlier, after a certain point changing a lot of things was also not helpful that much. The thing is it is asking to resize and change the format of images and we have used a lot of images. So, I worked on the ones that were really heavy and were affecting the performance greatly. 

So, I think pointlessly making any more image compressions or putting a lot of time into scaling images is fruitful only if we believe that would make a great difference. 
That's why I believe some review would be really helpful before proceeding.

Akshun-01 commented:
@Yashsharma1911 
What needs to be done here? I have already discussed this in the weekly meet twice now. 
And someone was required to review this and update with anything that needs to be done. 

If you or anyone else could take out some time. It would be great!
I have given in depth report for every change I made, so anyone can refer to it to decide what needs to be done.

Yashsharma1911 commented:
Great thanks then, I'll look in the evening, till I'll suggest you to parallel work on others issues too if you want as this seems to block you.

Message from mobile

Akshun-01 commented:
> Great thanks then, I'll look in the evening, till I'll suggest you to parallel work on others issues too if you want as this seems to block you.
> 
> Message from mobile

Yeah, I was stuck with a lot lately. So didn't get time. I will start with other issues as well now. 
Do you have any in mind, I would love to help!

Also,  I wanted to help in the backend part too, so if you know any beginner-friendly ones, I would appreciate it!
Thanks ^^

Yashsharma1911 reviewed (commented):
@Akshun-01 quality of this homepage GIF becomes too poor

![Screenshot 2024-02-12 at 1 10 11 PM](https://github.com/meshery/meshery.io/assets/71271069/a28d849f-5dc2-45f1-a09f-ef73e1ea24ff)

Akshun-01 commented:
> @Akshun-01 quality of this homepage GIF becomes too poor
> 
> ![Screenshot 2024-02-12 at 1 10 11 PM](https://private-user-images.githubusercontent.com/71271069/303989601-a28d849f-5dc2-45f1-a09f-ef73e1ea24ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3MzA3NzgsIm5iZiI6MTcwNzczMDQ3OCwicGF0aCI6Ii83MTI3MTA2OS8zMDM5ODk2MDEtYTI4ZDg0OWYtNWRjMi00NWYxLWEwOWYtZWY3M2UxZWEyNGZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEyVDA5MzQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzZTVkZTgxOGRjNmMyMTc3ZWE0M2JiYTczMmNkMDlkNTQzNGJlZWJmYmQ4ODcwZGY2MmExNmMxMTIzZmE3YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WENXCj-shONOuz1sbLgNLjGWPPQPxsYlVTqQkhYUcK0)

Yeah, it's not good. 
One thing I can do is. I now know what things help the most in performance, so I can close this one and work on the latest branch and improve it there. 
I will only make the necessary changes in that rather than optimizing and resizing every image.
What's your say?

sudhanshutech commented:
sounds helpful @Akshun-01 , before creating a pr it would be good if you have documented the changes first like what is the issues and what is the solution and series them according them to priority. So in this way it will not take that much time and will easy to review.
